gitpod-io/gitpod
# Support validate `claims` with CEL expr for SSO

## Description

<img width="385" alt="image" src="https://github.com/user-attachments/assets/a92a1cdc-b8db-4c16-a3ea-b77ee88e6651">


## Related Issue(s)

Relates ENT-561

## How to test

Current CEL expression in preview env is `'gitpod-team' in claims.groups_direct || (claims.email_verified && claims.email.endsWith("@gitpod.io"))`

- Log into preview env with GitLab.com x SSO, conditions below depends whether you could login successfully:
   - If your validated email is ends with `@gitpod.io`, you should be able to login
   - If your are a member of group `gitpod-team`, you should be able to login

**Setup Preview envs with Self-hosted / SaaS GitLab**

> [!NOTE]
> Test should follow https://github.com/gitpod-io/website/pull/4789

- Re trigger GHA if preview env is deleted, (or delete preview env when needed)
- Login with OIDC setup
```
previewctl admin credentials create
kubectl rollout restart deploy/server
```
- Setup OIDC with gitlab.com and Self-Hosted gitlab
- Play with CEL expression

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - hw-oidc-cel</li>
	<li><b>🔗 URL</b> - <a href="https://hw-oidc-cel.preview.gitpod-dev.com/workspaces" target="_blank">hw-oidc-cel.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - hw-oidc-cel-gha.27820</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-hw-oidc-cel%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
